src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -630,6 +631,10 @@ void Simulator::run()
 	_myaddr.sin_addr.s_addr = htonl(INADDR_ANY);
 	_myaddr.sin_port = htons(_port);
 
+	if (_ip == InternetProtocol::TCP_REMOTE) {
```

Comment:
And then here you can check for that:

```suggestion
	if (_tcp_remote_ipaddr != nullptr) {
```

That way the enum state and the `_tcp_remote_ipaddr` can't get out of sync.